src/types/artifact.ts

```diff
@@ -0,0 +1,21 @@
+import { ArtifactType, artifactTypes, ResultArtifactData, MarkdownArtifactData, Artifact, ArtifactData, TableArtifactData } from '@/models/Artifact'
+
+export function isArtifactType<T extends ArtifactType>(artifact: Artifact, type: T): artifact is Artifact & { type: T } {
+  return artifact.type === type
+}
+
+export function isKnownArtifactType(type: unknown): type is ArtifactType & Exclude<ArtifactType, 'unknown'> {
+  return typeof type === 'string' && artifactTypes.includes(type as ArtifactType) && type !== 'unknown'
```

Comment:
I thought the mapper ensured it was always an ArtifactType? So I expected this to not need the string type check or the includes.

Reply:
yea the mapper does but I'm not sure a typeguard should presume the existence of a mapper, especially when we'd be passing in a primitive either way (the type string). If we were passing in a fully-hydrated artifact object i don't think the full type check would make sense but in that case the narrowing has already been done

Reply:
Oh duh cause it accepts an unknown value. In that case makes perfect sense. 

I think the return type can remove the first `ArtifactType &`